### ISSUE TITLE
Problem with default emails after installing this package

### ISSUE THREAD

**Issue by jcerdan** [BJNSTNKVC/laravel-mail-components#5]
Hi,

I have installed this package on Laravel 11.x with Livewire and Fluxui.
After installing it, default emails for password recovery for example have bad css. Buttons doesn't appear and sometimes text is unreadable.

I could solve major issues setting prefix to another value to not conflict with default Laravel prefix.

```
// Mail Components prefix (E.g. <x-mail-button>).
'prefix'     => 'othervalue',
```

But buttons doesn't appear as you can see in the image:
![imagen](https://github.com/user-attachments/assets/5bcca8d3-8a80-4e91-b12f-ffcc57adce38)

Any ideas on how to solve this?

Regards

**Comment by BJNSTNKVC** [comment]
Hello Jordi, thanks for showing interest in the library. 

Have you cleared application cache after changin the prefix by running ```php artisan cache:clear```?

**Comment by jcerdan** [comment]
Hello Bojan,

Yes, I have run ```php artisan cache:clear``` and ```php artisan view:clear```

But still button does not appear...

?¿?¿

**Comment by BJNSTNKVC** [comment]
Can you share the contents of the following files:
1. mails_config.php
2. Mailable class
3. Mailable blade.php
4. Button class (if you have published it)
5. Button blade.php (if you have published it)

It would be even better if you can create a repo in which you can reproduce the issue and add me so I can pull it and see what's happening.

**Comment by jcerdan** [comment]
Hi Bojan,

here's the  mail_components.php.
Mail are sent by SMTp, and config/mail.php has not been changed.
About Mailable class it's default reset password Laravel Mail. 
No mail or notifications componenst have been published.

```
<?php

return [

    /*
    |--------------------------------------------------------------------------
    | Mail Components
    |--------------------------------------------------------------------------
    |
    | Here you may specify which components you would like to use as well
    | as their default behaviours.
    |
    */

    // List of all Components that are going to be loaded. Feel free to comment out the components you don't need.
    'components' => [
        'button'   => Bjnstnkvc\MailComponents\View\Components\Mail\Button::class,
        'content'  => Bjnstnkvc\MailComponents\View\Components\Mail\Content::class,
        'footer'   => Bjnstnkvc\MailComponents\View\Components\Mail\Footer::class,
        'grid'     => Bjnstnkvc\MailComponents\View\Components\Mail\Grid::class,
        'header'   => Bjnstnkvc\MailComponents\View\Components\Mail\Header::class,
        'hero'     => Bjnstnkvc\MailComponents\View\Components\Mail\Hero::class,
        'image'    => Bjnstnkvc\MailComponents\View\Components\Mail\Image::class,
        'layout'   => Bjnstnkvc\MailComponents\View\Components\Mail\Layout::class,
        'new-line' => Bjnstnkvc\MailComponents\View\Components\Mail\NewLine::class,
        'subcopy'  => Bjnstnkvc\MailComponents\View\Components\Mail\Subcopy::class,
        'table'    => Bjnstnkvc\MailComponents\View\Components\Mail\Table::class,
    ],

    // Mail Components prefix (E.g. <x-mail-button>).
    'prefix'     => 'tcbmail',

    // Mail Components separator (E.g. <x-mail::button>).
    'separator'  => '::',

    // Mail Layout component settings.
    'layout'     => [
        //'font_link'   => 'https://fonts.googleapis.com/css2?family=Roboto&display=swap',
        'font_family' => 'Roboto',
        'background'  => '#F2F2F2',
    ],

    // Mail Header component settings.
    'header'     => [
        'show'    => true,
        'app_url' => config('app.url'),
        'logo'    => '<path_to_public_logo>',
        'width'   => 262,
        'height'  => 75,
    ],

    // Mail Footer component info.
    'footer'     => [
        'show'           => true,
        'show_copyright' => true,
        'address'        => true,
        'city'           => true,
        'state'          => true,
        'zip'            => true,
        'phone'          => true,
    ],

    // Mail Content component settings.
    'content'    => [
        'background' => 'transparent',
    ],

    // Mail Grid component settings.
    'grid'       => [
        'background' => 'transparent',
        'spacing'    => 'top, bottom',
    ],

    // Mail Hero component settings.
    'hero'       => [
        'background' => 'transparent',
        'height'     => 400,
    ],

    // Mail Subcopy component settings.
    'subcopy'    => [
        'background' => 'transparent',
    ],

    // Mail Table component settings.
    'table'      => [
        'row_index'         => false,
        'background'        => 'rgba(255, 255, 255, 1)',
        'header_background' => 'rgba(115, 111, 110, 0.15)',
    ],

    // Mail Image component settings.
    'image'      => [
        'is_responsive' => false,
        'as_section'    => true,
    ],

    // Mail Button component settings.
    'button'     => [
        'width'  => 200,
        'height' => 40,
        'color'  => 'black',
    ],

    // Mail New Line component settings.
    'new_line'   => [
        'height'     => 16,
        'background' => 'transparent',
    ],
];
```

I suspect that maybe fluxui, wireui and tailwind with your css are somehow making an explosive melting pot here.
I'll try to install a clean Laravel with your package and see what happens.
Then I will add progressively other packages and see which one causes this.

Thanks,

**Comment by BJNSTNKVC** [comment]
I believe I found the source of the issue. Essentially, [Laravel Starter Kits](https://laravel.com/docs/11.x/starter-kits#main-content) are using [Laravel Notifcations ](https://laravel.com/docs/11.x/notifications) to create emails.

In this particular case, Password Reset Link is being sent via [this](https://github.com/laravel/breeze/blob/2.x/stubs/default/app/Http/Controllers/Auth/PasswordResetLinkController.php#L35) piece of code. If you follow through this method, we end up in [ResetPassword ](https://github.com/laravel/framework/blob/11.x/src/Illuminate/Auth/Notifications/ResetPassword.php) class which builds a message [here](https://github.com/laravel/framework/blob/11.x/src/Illuminate/Auth/Notifications/ResetPassword.php#L75).

In order to have Notifications use the components/styling from the library, we first need to publish the default Notifications template, as noted in the [docs](https://laravel.com/docs/11.x/notifications#customizing-the-templates).

Once published, [email.blade.php](https://github.com/laravel/framework/blob/11.x/src/Illuminate/Notifications/resources/views/email.blade.php) file can be found in resources/views/vendor/notifications folder.  In order to have library styles applied, you need to either globaly set the mail theme, as explained [here](https://github.com/BJNSTNKVC/laravel-mail-components#theme), or call [theme](https://github.com/laravel/framework/blob/11.x/src/Illuminate/Notifications/Messages/MailMessage.php#L183) method and pass the `mail_components` as a name.

```php
    /**
     * Get the reset password notification mail message for the given URL.
     *
     * @param  string  $url
     * @return \Illuminate\Notifications\Messages\MailMessage
     */
    protected function buildMailMessage($url)
    {
        $message =  (new MailMessage)
            ->theme('mail-components')
            ->subject(Lang::get('Reset Password Notification'))
            ->line(Lang::get('You are receiving this email because we received a password reset request for your account.'))
            ->action(Lang::get('Reset Password'), $url)
            ->line(Lang::get('This password reset link will expire in :count minutes.', ['count' => config('auth.passwords.'.config('auth.defaults.passwords').'.expire')]))
            ->line(Lang::get('If you did not request a password reset, no further action is required.'));
    }
```

The latter in this particular case is probably not a good idea as aforementioned method is found in the vendor folder, however, you can customize the notifications as described [here](https://laravel.com/docs/11.x/passwords#reset-email-customization).

Last piece of the puzzle is to actually modify the email.blade.php file to suite your needs. For example, this can be a good starting point:

```blade
<x-mail::layout>
    <x-mail::header logo="https://laravel.com/img/notification-logo.png" />

    <x-mail::content background="#FFFFFF">
        {{-- Greeting --}}
        @if (! empty($greeting))
            <h1>{{ $greeting }}</h1>
        @else
            <h1>
                @if ($level === 'error')
                    @lang('Whoops!')
                @else
                    @lang('Hello!')
                @endif
            </h1>
        @endif

        {{-- Intro Lines --}}
        @foreach ($introLines as $line)
            {{ $line }}
        @endforeach

        <x-mail::new-line as-table />

        {{-- Action Button --}}
        @isset($actionText)
                <?php
                $color = match ($level) {
                    'success', 'error' => $level,
                    default            => 'black',
                };
                ?>
            <x-mail::button :url="$actionUrl" :color="$color">{{ $actionText }}</x-mail::button>
        @endisset

        <x-mail::new-line as-table />

        {{-- Outro Lines --}}
        @foreach ($outroLines as $line)
            {{ $line }}
        @endforeach

        <x-mail::new-line as-table />

        {{-- Salutation --}}
        @if (! empty($salutation))
            {{ $salutation }}
        @else
            @lang('Regards,')
            {{ config('app.name') }}
        @endif
    </x-mail::content>

    {{-- Subcopy --}}
    @isset($actionText)
        <x-mail::subcopy>
            @lang("If you're having trouble clicking the \":actionText\" button, copy and paste the URL below\n". 'into your web browser:', [
                'actionText' => $actionText,
            ])
            <span class="break-all">{!! Illuminate\Support\Str::markdown("[$displayableActionUrl]($actionUrl)") !!}</span>
        </x-mail::subcopy>
    @endisset

    <x-mail::footer />
</x-mail::layout>
```
Please note that [new line](https://github.com/BJNSTNKVC/laravel-mail-components#new-line) component uses `as-table` attribute which is available with the latest version ([1.1.5](https://github.com/BJNSTNKVC/laravel-mail-components/releases/tag/1.1.5)), so update the library if you haven't already.

When received, email will look as follows:
![Screenshot 2024-11-25 211950](https://github.com/user-attachments/assets/b3227081-8890-458c-82a7-5a4916891c39)

**Comment by BJNSTNKVC** [comment]
I'm closing this issue. Feel free to reopen it in case you need any assistance or find something out of the ordinary.
